src/main/java/com/ldtteam/blockout/views/TabSet.java

```diff
@@ -0,0 +1,271 @@
+        }
+        else
+        {
+            tabs.setPosition(offset, overlap - tabHeight);
+            tabs.setSize(width - 2*offset, tabHeight);
+        }
+
+        spacing = params.getInteger("spacing", 2);
+    }
+
+    /**
+     * Selects a tab from its index in the tab button list
+     * @param index the index to select
+     */
+    public void setSelectedTab(int index)
+    {
+        if (index >= tabs.children.size()) return;
+        Pane tab = tabs.children.get(index);
+        if (!(tab instanceof ButtonTab)) return;
+        selectedTab = ((ButtonTab) tab).toSelect;
+
+        children.stream()
+          .filter(child -> child instanceof Tab)
+          .forEach(child -> child.setVisible(child.equals(selectedTab)));
+    }
+
+    /**
+     * Changes the selected tab view to the given one
+     * @param tab the tab to switch to
+     */
+    public void setSelectedTab(Tab tab)
+    {
+        children.stream()
+          .filter(child -> child instanceof Tab)
+          .forEach(child -> {
+              child.setVisible(child.equals(tab));
+              if (child.isVisible()) selectedTab = tab;
+          });
+    }
+
+    /**
+     * Adds the given tab to this tab set
+     * @param tab the tab to add
+     */
+    public void addTab(Tab tab)
+    {
+        tab.padding = childPadding;
+        addChild(tab);
+        new ButtonTab(tab);
+    }
+
+    /**
+     * Adds the given tab to this tab set
+     * @param tab the tab to add
+     * @param button an existing ButtonTab to link to the given tab
+     */
+    public void addTab(Tab tab, ButtonTab button)
+    {
+        tab.padding = childPadding;
+        button.toSelect = tab;
+        addChild(tab);
+    }
+
+    @Override
+    public void parseChildren(@NotNull final PaneParams params)
+    {
+        super.parseChildren(params);
+
+        getChildren().stream()
+          .filter(child -> child instanceof Tab)
+          .map(child -> (Tab) child)
+          .peek(child -> child.padding = childPadding)
+          .forEach(ButtonTab::new);
+
+        setSelectedTab(0);
+    }
+
+    @Override
+    protected boolean childIsVisible(final Pane child)
+    {
+        // Draw the tabs despite being potentially out of the tab set view
+        return child.getID().equals("tabs") || super.childIsVisible(child);
+    }
+
+    @Override
+    public boolean isPointInPane(final double mx, final double my)
+    {
+        // Essentially resize the 'bounding box' for events
+        return isVisible() && vertical
+            ? mx >= x + overlap - tabWidth && mx < (x + width) && my >= y && my < (y + height)
+            : mx >= x && mx < (x + width) && my >= y + overlap - tabHeight && my < (y + height);
+    }
+
+    /**
+     * The pane for the physical representation of the tab itself,
+     * i.e. the button you click to change content.
+     * Has a background and a potential icon or text otherwise
+     */
+    public class ButtonTab extends View
```

Comment:
suggestion: TabButton?

Reply:
Another thing, this should extend Button, not View, because of disabled/hover images etc. so one can use disabled image for tab button being selected or not (propably use Pane.enabled to disable when selected)